### PR TITLE
Update versions of plugins and checkstyle too

### PR DESCRIPTION
**Pull request by nzroller**
Updates plugins to the latest released. This includes a couple of
"milestone" releases (javadoc and enforcer). Checkstyle updated to v8.2.

### CODE REVIEW

File: basepom/pom.xml [hunk 1]
```diff
@@ -32,28 +32,28 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <buildNumber>${project.version}</buildNumber>
-        <checkstyle.version>7.1.2</checkstyle.version>
-        <dependency-check-maven.version>2.0.1</dependency-check-maven.version>
+        <checkstyle.version>8.2</checkstyle.version>
```
Comment (xiaodong-izettle) [inline basepom/pom.xml]: @nzroller, it seems 8.3 is out already. 😄

Comment (xiaodong-izettle) [reply basepom/pom.xml]: I upgraded `checkstyle` to 8.3, and tweaked the `checkstyle` settings to confirm with what we have right now, @nzroller

File: basepom/pom.xml [hunk 2]
```diff
@@ -32,28 +32,28 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <buildNumber>${project.version}</buildNumber>
-        <checkstyle.version>7.1.2</checkstyle.version>
-        <dependency-check-maven.version>2.0.1</dependency-check-maven.version>
-        <izettle-checkstyle.version>2.1.17</izettle-checkstyle.version>
+        <checkstyle.version>8.3</checkstyle.version>
+        <dependency-check-maven.version>2.1.1</dependency-check-maven.version>
+        <izettle-checkstyle.version>2.1.18</izettle-checkstyle.version>
```
Comment (xiaodong-izettle) [inline basepom/pom.xml]: Need a new release of `izettle-checkstyle` for this to work of course.

File: osspom/pom.xml [hunk 3]
```diff
@@ -60,30 +60,30 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
         <buildNumber>${project.version}</buildNumber>
-        <checkstyle.version>7.1.2</checkstyle.version>
-        <dependency-check-maven.version>1.4.5</dependency-check-maven.version>
-        <izettle-checkstyle.version>2.1.17</izettle-checkstyle.version>
+        <checkstyle.version>8.2</checkstyle.version>
+        <dependency-check-maven.version>2.1.1</dependency-check-maven.version>
+        <izettle-checkstyle.version>2.1.18</izettle-checkstyle.version>
```
Comment (xiaodong-izettle) [inline osspom/pom.xml]: Need a new release of izettle-checkstyle for this to work of course.

File: checkstyle/src/main/resources/checkstyle/checkstyle.xml [hunk 4]
```diff
@@ -159,6 +159,7 @@
         <module name="ImportOrder">
             <property name="option" value="top"/>
             <property name="ordered" value="true"/>
+            <property name="separated" value="true"/>
```
Comment (xiaodong-izettle) [inline checkstyle/src/main/resources/checkstyle/checkstyle.xml]: For default Intellij setting

**Comment by adamvoncorswant** [comment]
will this make us able to start use java9?

**Comment by xiaodong-izettle** [comment]
Necessary but not enough, @adamvoncorswant 😄

**Comment by xiaodong-izettle** [comment]
Nice one, @nzroller. 👍

**Pull request merged**